ui/src/app/machines/views/MachineDetails/MachineLogs/DownloadMenu/DownloadMenu.test.tsx

```diff
@@ -63,6 +78,161 @@ describe("DownloadMenu", () => {
+          initialEntries={[{ pathname: "/machines", key: "testKey" }]}
+        >
+          <DownloadMenu systemId="abc123" />
+        </MemoryRouter>
+      </Provider>
+    );
+    expect(wrapper.find("ContextualMenu").prop("toggleDisabled")).toBe(true);
+  });
+
+  it("does not display a YAML output item when it does not exist", () => {
+    const store = mockStore(state);
+    const wrapper = mount(
+      <Provider store={store}>
+        <MemoryRouter
+          initialEntries={[{ pathname: "/machines", key: "testKey" }]}
+        >
+          <DownloadMenu systemId="abc123" />
+        </MemoryRouter>
+      </Provider>
+    );
+    expect(
+      wrapper
+        .find("ContextualMenu")
+        .prop("links")
+        .some(({ children }) => children === "Machine output (YAML)")
+    ).toBe(false);
+  });
+
+  it("can display a YAML output item", () => {
+    jest.spyOn(fileContextStore, "get").mockReturnValue("test yaml file");
+    const store = mockStore(state);
+    const wrapper = mount(
+      <Provider store={store}>
+        <MemoryRouter
+          initialEntries={[{ pathname: "/machines", key: "testKey" }]}
+        >
+          <FileContext.Provider value={fileContextStore}>
+            <DownloadMenu systemId="abc123" />
+          </FileContext.Provider>
+        </MemoryRouter>
+      </Provider>
+    );
+    expect(
+      wrapper
+        .find("ContextualMenu")
+        .prop("links")
+        .some(({ children }) => children === "Machine output (YAML)")
+    ).toBe(true);
+  });
+
+  it("generates a download when the installation item is clicked", () => {
+    jest.spyOn(fileContextStore, "get").mockReturnValue("test yaml file");
+    jest
+      .useFakeTimers("modern")
+      .setSystemTime(new Date("2021-03-25").getTime());
+    const downloadSpy = jest.spyOn(fileDownload, "default");
+    const store = mockStore(state);
+    const wrapper = mount(
+      <Provider store={store}>
+        <MemoryRouter
+          initialEntries={[{ pathname: "/machines", key: "testKey" }]}
+        >
+          <DownloadMenu systemId="abc123" />
+        </MemoryRouter>
+      </Provider>
+    );
+    wrapper
+      .find("ContextualMenu")
+      .prop("links")
+      .find(({ children }) => children === "Machine output (YAML)")
+      .onClick();
+    expect(downloadSpy).toHaveBeenCalledWith(
+      "test yaml file",
+      "hungry-wombat.aus-machine-output-2021-03-25.yaml"
+    );
+  });
+
+  it("does not display a XML output item when it does not exist", () => {
+    const store = mockStore(state);
+    const wrapper = mount(
+      <Provider store={store}>
+        <MemoryRouter
+          initialEntries={[{ pathname: "/machines", key: "testKey" }]}
+        >
+          <DownloadMenu systemId="abc123" />
+        </MemoryRouter>
+      </Provider>
+    );
+    expect(
+      wrapper
+        .find("ContextualMenu")
+        .prop("links")
+        .some(({ children }) => children === "Machine output (XML)")
```

Comment:
There are a number of these, can you please update all. This may be as simple to solve as adding a prop `type` on the children to inspect vs the display string.

Reply:
I see you're already passing the `extension` to the item generation so that can be exposed as a prop.